src/vision/src/HSVFilterNode.cpp

```diff
@@ -6,23 +6,81 @@
+ * wroking yaml file.
+ */
+void mySigintHandler(int sig) {
+    std::cout << "\nUpdating the yaml files... preparing for shutdown.\n"
+              << std::endl;
+    std::string path = ros::package::getPath("vision");
+
+    // Create the terminal call to run the python executable
+    std::string path_to_script    = path + "/cfg/yamlEditor.py ";
+    std::string path_to_yaml_file = path + "/cfg/rqt_params_out.yaml \"";
+    std::string path_to_out_file  = path + "/cfg/launch_params_out.yaml\"";
+    std::string command =
+    "python " + path_to_script + path_to_yaml_file + path_to_out_file;
+
+    // Run the python script to clean up yaml files
+    system(("chmod +x " + path_to_script).c_str());
+    system(command.c_str());
+
+    // All the default sigint handler does is call shutdown()
+    ros::shutdown();
+}
+
 HSVFilterNode::HSVFilterNode(int argc, char** argv, std::string node_name) {
     // Setup NodeHandles
-    ros::init(argc, argv, node_name);
+    ros::init(
+    argc, argv, node_name, ros::init_options::NoSigintHandler); // added
     ros::NodeHandle nh;
     ros::NodeHandle private_nh("~");
     image_transport::ImageTransport it(nh);
 
     std::string subscribeTopic = "/camera/image_raw";
     std::string publishTopic   = "/vision/output";
 
-    dynamic_reconfigure::Server<vision::hsvfilterConfig> server;
-    dynamic_reconfigure::Server<vision::hsvfilterConfig>::CallbackType f;
+    // Bool to retrieve from the param-server that was loaded in the launch
+    // file.
+    bool is_dyn_recon;
+
+    // the parameters we care about to be pulled from rosparam server
+    int h_high, h_low, v_high, v_low, s_high, s_low;
 
-    filter_ = HSVFilter();
+    // Load all the parameters from the rosparam server
+    nh.getParam("/" + node_name + "/h_high", h_high);
+    nh.getParam("/" + node_name + "/h_low", h_low);
+    nh.getParam("/" + node_name + "/v_high", v_high);
+    nh.getParam("/" + node_name + "/v_low", v_low);
+    nh.getParam("/" + node_name + "/s_high", s_high);
+    nh.getParam("/" + node_name + "/s_low", s_low);
+    nh.getParam("/" + node_name + "/is_dyn_recon", is_dyn_recon);
 
-    f = boost::bind(&HSVFilterNode::dynamicreconfigCallback, this, _1, _2);
-    server.setCallback(f);
+    // Create a new filter object with the values pulled from the rosparam
+    // server (which is configured in the launch file)
+    filter_ = HSVFilter(h_low, h_high, s_low, s_high, v_low, v_high);
+
+    // If this line is inside the if statement, this node doesnt show up on the
+    // dynamic reconfigure GUI
+    dynamic_reconfigure::Server<vision::hsvfilterConfig> server;
+
+    if (is_dyn_recon) {
+        dynamic_reconfigure::Server<vision::hsvfilterConfig>::CallbackType f;
+        f = boost::bind(&HSVFilterNode::dynamicreconfigCallback, this, _1, _2);
+        server.setCallback(f);
+    } else {
+        // If the Dynamic Reconfigure is not desired, then set all the
```

Comment:
Do we need this resetting of the parameters? If not can we take it out

Reply:
It is not needed. I only added it there so that the values in the "params server" would be consistent with the actual values being used in the node. In terms of functionality, it is not necessary. Feel free to take out the code within that else block.